app/src/main/java/de/westnordost/streetcomplete/user/LoginFragment.kt

```diff
@@ -137,7 +137,9 @@ class LoginFragment : Fragment(R.layout.fragment_login),
         private val REQUIRED_OSM_PERMISSIONS = listOf(
             Permission.READ_PREFERENCES_AND_USER_DETAILS,
             Permission.MODIFY_MAP,
-            Permission.WRITE_NOTES
+            Permission.WRITE_NOTES,
+            Permission.READ_GPS_TRACES,
+            Permission.WRITE_GPS_TRACES
```

Comment:
Currently if a user is already login they can upload existing quests as is, nothing changes here.

(after the below committed refactor):
If a login user creates a track and tries to sync it, it will have a popup saying they need to re-authorize due to throwing `OsmAuthorizationException`. This means that login users will only need to "upgrade" their permissions after they create one of these. New users should already have this permission requested (since there is only one app / client id...).
![image](https://user-images.githubusercontent.com/2222562/145932260-24e22b0c-587b-44a6-8a75-3b94dec216ff.png)

Reply:
Hmm, so this forcibly logs out the user when he firet uses this feature. This behavior will look as if it is a bug.

Reply:
I added a custom dialog which informs the user why they are being logged out.
Not sure how to handle it without logging out the user back out so that they can re-request with all the correct permissions.
![image](https://user-images.githubusercontent.com/2222562/146634077-8ab94521-131a-462e-8362-ac432b30a875.png)

This will only get shown if a current user uploads a trace, otherwise they won't see this.
New users will always have all correct permissions after their first login.